src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php

```diff
@@ -153,13 +153,34 @@ protected function addCookieToResponse($request, $response)
     {
         $config = config('session');
 
-        $response->headers->setCookie(
-            new Cookie(
-                'XSRF-TOKEN', $request->session()->token(), Carbon::now()->getTimestamp() + 60 * $config['lifetime'],
-                $config['path'], $config['domain'], $config['secure'], false
-            )
-        );
+        $isXSRFDeleted = $this->isXSRFDeleted($response->headers->getCookies());
+        if (! $isXSRFDeleted) {
```

Comment:
You could probably merge these two lines into one because the variable is only referenced here and the method call is fair self-explanatory